api/api/models/media.py

```diff
@@ -183,60 +181,111 @@ class AbstractMediaReport(models.Model):
         help_text="The explanation on why media is being reported.",
     )
     status = models.CharField(max_length=20, choices=STATUS_CHOICES, default=PENDING)
+    """
+    All statuses except ``PENDING`` are deprecated. Instead refer to the
+    properties ``is_pending`` and ``is_moderated``.
+    """
+
+    decision = models.ForeignKey(
+        to="AbstractMediaDecision",
+        on_delete=models.SET_NULL,
+        blank=True,
+        null=True,
+        help_text="The moderation decision for this report.",
+    )
 
     class Meta:
         abstract = True
 
     def clean(self):
         """Clean fields and raise errors that can be handled by Django Admin."""
 
-        if not self.media_class.objects.filter(
-            identifier=self.media_obj.identifier
-        ).exists():
+        if not self.media_class.objects.filter(identifier=self.media_obj_id).exists():
             raise ValidationError(
-                f"No '{self.media_class.__name__}' instance"
-                f"with identifier {self.media_obj.identifier}."
+                f"No '{self.media_class.__name__}' instance "
+                f"with identifier '{self.media_obj_id}'."
             )
 
-    def url(self, media_type):
-        url = f"{settings.CANONICAL_ORIGIN}v1/{media_type}/{self.media_obj.identifier}"
+    @property
+    def url(self):
+        origin = settings.CANONICAL_ORIGIN
+        if not origin.endswith("/"):
+            origin = f"{origin}/"
+        url = f"{origin}v1/{self.url_frag}/{self.media_obj.identifier}"
```

Comment:
This should use reverse: https://docs.djangoproject.com/en/5.0/ref/urlresolvers/#reverse

Is there a need to construct this manually? This manual constructed version is also missing the trailing slash, which reverse should include based on our configuration.

We should also change this to `media_object_url` or something more descriptive of the target of the URL. It took me a while to register the fact that this isn't the url for the report, and I only realised because I ended up spending a couple minutes trying to understand what the report URL would be used for.

This also should not dot-through to `media_obj.identifier` as it will cause an unnecessary DB query. `media_obj_id` can be used instead to avoid the join.


```suggestion
    def media_obj_url(self):
        return reverse("image-detail", args=[self.media_obj_id])
```

This won't give an absolute URL, but the admin can (and should) generate that based on the request, using `request.get_absolute_uri` and pass in the `.media_obj_url` from the report instance.

If we don't do that, then if someone is on `api-production.openverse.org` and clicks on one of these urls, their session will get trashed. They'll need to re-login through Cloudflare access and then again at Django admin, and at that point probably have lost the `redirect_to` query parameter, but at the very least, their work was unnecessarily disrupted.

Reply:
I've updated the code to use `reverse` and generate only links with paths in 85e1678. That way the links will not point to a different domain and we won't have to worry about sessions.

![localhost_50280_admin_api_imagereport__q=2e22](https://github.com/WordPress/openverse/assets/16580576/2a578916-0f63-48a0-86f7-e2bae3022adf)